arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/Hub.java

```diff
@@ -362,95 +354,160 @@ public List<Module> getModulesToTrace() {
    *
    * @return the modules to count
    */
-  public List<Module> getModulesToCount() {
+  public List<Module> getDefaultModulesToCount() {
     return Stream.concat(
-            Stream.of(
-                this,
-                add,
-                bin,
-                blakeModexpData,
-                blockdata,
-                blockhash,
-                ecData,
-                exp,
-                ext,
-                euc,
-                gas,
-                logData,
-                logInfo,
-                mmu,
-                mmio,
-                mod,
-                mul,
-                mxp,
-                oob,
-                rlpAddr,
-                rlpTxn,
-                rlpTxnRcpt,
-                rom,
-                romLex,
-                shakiraData,
-                shf,
-                stp,
-                trm,
-                txnData,
-                wcp,
-                l2Block,
-                l2L1Logs),
-            Stream.concat(refTableModules.stream(), precompileLimitModules().stream()))
+            standardModulesToCount.stream(),
+            Stream.concat(refTableModules.stream(), precompileLimitModules.stream()))
         .toList();
   }
 
-  public Hub(
-      final Address l2l1ContractAddress,
-      final Bytes l2l1Topic,
-      final BigInteger nonnegativeChainId) {
-    Preconditions.checkState(nonnegativeChainId.signum() >= 0);
-    chainId = nonnegativeChainId;
-    l2Block = new L2Block(l2l1ContractAddress, LogTopic.of(l2l1Topic));
-    l2L1Logs = new L2L1Logs(l2Block);
-    keccak = new Keccak(ecRecoverEffectiveCall, l2Block);
-    shakiraData = new ShakiraData(wcp, sha256Blocks, keccak, ripemdBlocks);
-    blockdata = new Blockdata(wcp, txnData, rlpTxn, chainId);
-    mmu = new Mmu(euc, wcp);
-    mmio = new Mmio(mmu);
+  private List<Module> initStandardModulesToCount() {
+    return List.of(
+        this,
+        add,
+        bin,
+        blakeModexpData,
+        blockdata,
+        blockhash,
+        ecData,
+        exp,
+        ext,
+        euc,
+        gas,
+        logData,
+        logInfo,
+        mmu,
+        mmio,
+        mod,
+        mul,
+        mxp,
+        oob,
+        rlpAddr,
+        rlpTxn,
+        rlpTxnRcpt,
+        rom,
+        romLex,
+        shakiraData,
+        shf,
+        stp,
+        trm,
+        txnData,
+        wcp,
+        l2Block,
+        l2L1Logs);
+  }
+
+  private List<Module> initStandardModulesToTrace() {
```

Comment:
maybe, the standardModuleToTrace (and to Count) could be initStandardModule + some few others (just take care if some order are needed, like for mum/mmio at tracing time)

Reply:
I was trying to do that but I noticed one issue. `StandardModulesToTrace` does not have `gas` but both `StandardModules` and `StandardModulesToCount` have it. I am not sure is this is correct but it this is the expected case then we are removing a module from `StandardModules` to get `StandardModulesToTrace`.

Another issue that can arise is it would be hard to enforce an order if I do list operations because it will be very difficult to position the extra modules in an already existing list of `StandardModules`